planner/cascades/transformation_rules.go

```diff
@@ -649,6 +652,65 @@ func (r *PushSelDownWindow) OnTransform(old *memo.ExprIter) (newExprs []*memo.Gr
 	return []*memo.GroupExpr{newTopSelExpr}, true, false, nil
 }
 
+// MergeAdjacentWindow merge adjacent Window.
+type MergeAdjacentWindow struct {
```

Comment:
Why we need it? They are already merged in plan build stage.

Reply:
PlanBuilder only merges Windows if we have defined a `Window` in the SQL, like:
```sql
select sum(a) over w, max(a) over w from t window w as (partition by b order by c)
```

But it cannot handle this scenario:
```sql
select sum(a) over (partition by b order by c), max(a) over (partition by b order by c) from t 
```

Reply:
Then why not just modify the plan builder?

Reply:
IMHO, using a rule could handle more general cases, like:
```sql
 select t2.*, max(a) over (partition by b) from (select t1.*, sum(a) over (partition by b) from t1) t2;
```